pylint/plugins/hass_imports.py

```diff
@@ -524,6 +546,19 @@ def visit_importfrom(self, node: nodes.ImportFrom) -> None:
                             node=node,
                             args=(import_match.string, obsolete_import.reason),
                         )
+        for name in node.names:
+            self._has_invalid_namespace_import(node, node.modname, name[0])
+
+    def _has_invalid_namespace_import(
+        self, node: nodes.ImportFrom, module: str, name: str
+    ) -> None:
+        for key, value in _FORCE_NAMESPACE_IMPORT.items():
+            if module == key[0] and name in value:
```

Comment:
It quite inefficient to iterate over all items in `_FORCE_NAMESPACE_IMPORT` for each `ImportFrom` node name. Let's change the data structure of `_FORCE_NAMESPACE_IMPORT` so we can use a dict lookup instead.

Maybe something like:
```py
class NamespaceAlias(NamedTuple):
    alias: str
    names: set[str]  # function names

_FORCE_NAMESPACE_IMPORT: dict[str, NamespaceAlias] = {...}
```

Reply:
Implemented